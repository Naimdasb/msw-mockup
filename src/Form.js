import { useState } from 'react';

export const Form = () => {
    const [username, setUsername] = useState('')
    const [password, setPassword] = useState('')
    const [response, setResponse] = useState({username: '', password: ''})


    const handleChange = (event) => {
        if(event.target.name === 'username') {
            setUsername(event.target.value)
        } else {
            setPassword(event.target.value)
        }
    }

    const handleSubmit = (event) => {
        fetch('/send', { method: 'POST',
                         headers: {'Content-Type': 'application/json'},
                         body: JSON.stringify({username, password}) 
                        }
                        )
            .then(res => res.json())
            .then(res => setResponse(res))
        event.preventDefault()
    }

    return (
        <form onSubmit={handleSubmit}>
            <h1>{response.username}</h1>
            <h2>{response.password}</h2>
            <label htmlFor='username'>Username</label>
            <br />
            <input onChange={handleChange} value={username} id='username' name='username' type='text' />
            <br />
            <label htmlFor='password'>Password</label>
            <br />
            <input onChange={handleChange} value={password} id='password' name='password' type='text' />
            <br />
            <button type='submit'>Submit</button>
        </form>
    )
}