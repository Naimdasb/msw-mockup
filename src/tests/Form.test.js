import { render, cleanup, fireEvent, waitFor } from '@testing-library/react';
import { Form } from '../Form';
import userEvent from '@testing-library/user-event'


afterEach(cleanup)

test('Form test', async () => {
    const { getByRole, getByLabelText, getByText } = render(<Form />)
    const usernameInput = getByLabelText('Username')
    const passwordInput = getByLabelText('Password')
    const submit = getByRole('button', {name: 'Submit'})
    expect(usernameInput).toBeInTheDocument()
    expect(passwordInput).toBeInTheDocument()
    expect(submit).toBeInTheDocument()

    userEvent.type(usernameInput,'John')
    expect(usernameInput).toHaveValue('John')

    userEvent.type(passwordInput,'Chon')
    expect(passwordInput).toHaveValue('Chon')

    userEvent.click(submit)

    await waitFor(() => getByText('John'))
    await waitFor(() => getByText('Chon'))


})

test('Form test error', async () => {
    const { getByRole, getByLabelText, getByText } = render(<Form />)
    const usernameInput = getByLabelText('Username')
    const passwordInput = getByLabelText('Password')
    const submit = getByRole('button', {name: 'Submit'})
    expect(usernameInput).toBeInTheDocument()
    expect(passwordInput).toBeInTheDocument()
    expect(submit).toBeInTheDocument()

    userEvent.type(usernameInput,'John')
    expect(usernameInput).toHaveValue('John')

    userEvent.type(passwordInput,'fail')
    expect(passwordInput).toHaveValue('fail')

    userEvent.click(submit)

    await waitFor(() => getByText('Error'))
   
})
