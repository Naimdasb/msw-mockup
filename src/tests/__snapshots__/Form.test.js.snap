// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test 1`] = `
<div>
  <form>
    <h1>
      John
    </h1>
    <h2>
      Chon
    </h2>
    <label
      for="username"
    >
      Username
    </label>
    <br />
    <input
      id="username"
      name="username"
      type="text"
      value="John"
    />
    <br />
    <label
      for="password"
    >
      Password
    </label>
    <br />
    <input
      id="password"
      name="password"
      type="text"
      value="Chon"
    />
    <br />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
